### ISSUE TITLE
Problems Decrypting two KFX books

### ISSUE THREAD

**Issue by ae0579** [apprenticeharper/DeDRM_tools#1738]
The problems started when my Kindle for Pc 1.17 suddenly began with errors and didn't let to register my user. After a lot of proves, y founded that version 1.25.1 works fine. So for a time I had successfully working DeDRM tool with Kindle, but two days ago I downloaded some books and two of them didn't decrypt. I tried again to install Kindle 1.17 (and any other version between it and 1.25) but it is impossible for me. So I take a look at the error log and found something related to Python3 (that I had not installed previously in Wine) so, installed it. I deleted the book, downloaded again and imported in Calibre, but again couldn't be decrypted. So, to aboid Python3, I downgraded Calibre to 4.23 and DeDRM 6.8.

The actual situation is:
Ubuntu 20.04.2
Genome Version: 3.36.8
Wine 6.01

The log is 
[tmpSyZRXX.txt](https://github.com/apprenticeharper/DeDRM_tools/files/6680630/tmpSyZRXX.txt)

And, for debugging needs, the book can be buyed (with cost of 0€) in
https://www.amazon.es/gp/product/B00B1W4A2K/ref=kinw_myk_ro_title

¿Is there any solution?

Thanks in advanced

Esther

**Comment by ElleKayEm** [comment]
Some people have reported that if you install the latest version of Kindle for PC, log in, then install Kindle for PC 1.17 over it without uninstalling the new version, that works.  Or you can disable KFX downloads using method 2 here:  https://www.mobileread.com/forums/showthread.php?t=283371

**Comment by ae0579** [comment]
True, and I tried both, but any works for me. I really don't know how to make 1.17 version work again

**Comment by ElleKayEm** [comment]
You'll need to use no later than Kindle for PC 1.26 or disable KFX downloads.  DeDRM cannot decrypt KFX format from Kindle for PC 1.27 or later.

**Comment by ae0579** [comment]
Sure, that is the reason I tried with version 1.25

**Comment by ElleKayEm** [comment]
The book in the log is from 1.25?  And that is the current version you have installed now?

**Comment by ae0579** [comment]
Yes.

**Comment by ElleKayEm** [comment]
DeDRM isn't finding the key.  You can try running kindlekey.py manually and importing the key to the plugin.  Or you could temporarily install Windows calibre and DeDRM where you have Kindle for PC.  Decrypt a book there, save the key, and import it into the Linux setup.

**Comment by ae0579** [comment]
But, It find the key when I download another book! And decript it with no problems! For example, I downloaded too the next book:

https://www.amazon.es/gp/product/B008EWNCC4/ref=kinw_myk_ro_title

And I think this book has DRM because I can't open it directly, but there is no error when I import it in Calibre, so DeDRM is working fine

**Comment by ElleKayEm** [comment]
That was downloaded with the current installation of Kindle for PC too?  Let's see a log.

**Comment by ae0579** [comment]
Yes, with the same installation, but now I have another problem. I'm stupid and tried again to reinstall Kindle, and now Kindle is not working (it gives me a C++ error when installing a version .25 or bigger and with previous version it don't connect to Internet). Yes, I could have been still.

**Comment by ElleKayEm** [comment]
I don't know how to fix that.  But the key changes with a new installation and log in.  So if you get things working again, you will need to re-download books which will work with the new key.

**Comment by ae0579** [comment]
Well, finally my vacancy began, and I had time to solve my problem with the installation of Kindle for Pc. Any way, previous versions of Kindle don't connect to Internet for me, so I have been forced to reinstal version 1.25.1 again.

The current situation is:

Ubuntu 20.04.2
Genome Version: 3.36.8
Wine 6.01
Kindle version: 1.25.1

For testing purpose, I downloaded 106 books and found that 65 could be added to Calibre perfectly in KFX format but the rest give errors. I Tried to avoid the use of KFX using the method explained with KFX Input and downloaded again one of them. It downloaded in azw format but DRM still persists.

The log is:

[Calibre log.txt](https://github.com/apprenticeharper/DeDRM_tools/files/6865372/Calibre.log.txt)

I looked the log and seems that DeDRM is not working, but today I deleted it, closed Calibre and reinstall the last version.

It seems to me that there is a new DRM, Could it be true or may be an installation issue?

Thanks in advance.

**Comment by ElleKayEm** [comment]
In the first log, DeDRM is unable to retrieve the key because it isn't finding Python installed under Wine.  In the second log, the book was already DRM-free.

**Comment by ae0579** [comment]
So, python 3 is a new requisite? It's true that now I don't have installed Python 3 but at the beginning I tried this and didn't work. I'll take a look when had some time. 
Thanks for your help!
